simulator/src/file/Open.js

```diff
@@ -0,0 +1,66 @@
+import load from '../data/load';
+import { generateSaveData } from '../data/save';
+
+const scopeSchema = ['layout', 'verilogMetadata', 'allNodes', 'id', 'name', 'restrictedCircuitElementsUsed', 'nodes'];
+const JSONSchema = ['name', 'timePeriod', 'clockEnabled', 'projectId', 'focussedCircuit', 'orderedTabs', 'scopes'];
+
+var circuitData = null;
+const GetDialogData = () => '<div><label for="CircuitDataFile">Choose file</label><div id="message-box"><i class="fas fa-plus"></i><br/>Browse files or Drag & Drop files here<div id="fileName">No file chosen!!</div><input style="background:none;" type="file" id="CircuitDataFile"/></div></div>';
+
+const ImportCircuitFiles = () => {
+    $('#ImportCircuitFilesDialog').empty();
+    $('#ImportCircuitFilesDialog').append(GetDialogData());
+    $('#ImportCircuitFilesDialog').dialog({
+        resizable: false,
+        close() {
+            if (circuitData) load(circuitData);
+        },
+        buttons: [
+            {
+                text: 'Close',
+                click() {
+                    $(this).dialog('close');
+                },
+            },
+        ],
+    });
+    $('#ImportCircuitFilesDialog').focus();
+
+    function ValidateData(circuitData) {
+        try {
+            circuitData = JSON.parse(circuitData);
+            if (JSON.stringify(Object.keys(circuitData)) !== JSON.stringify(JSONSchema)) throw new Error(`Invalid JSON data`);
+            circuitData.scopes.forEach((scope) => {
+                const keys = Object.keys(scope); // get scope keys
+                scopeSchema.forEach((key) => {
+                    if (!keys.includes(key)) throw new Error('Invalid Scope data');
+                });
+            });
+            load(circuitData);
+            return true;
+        } catch (error) {
+            $('#fileName').html(`<div class="text-danger">${error}</div>`);
+            return false;
+        }
+    }
+
+    function receivedText(e) {
+        // backUp data
+        const backUp = JSON.parse(generateSaveData($('#projectName').text().trim(), false));
+        const valid = ValidateData(e.target.result); // validate data
+        if (!valid) {
+            // fallback
+            load(backUp);
```

Comment:
fallback ... if any error occurs while loading file data... then it back up immediately.